master/internal/task/allocation.go

```diff
@@ -367,13 +377,19 @@ func (a *Allocation) RequestResources(ctx *actor.Context) error {
 // Cleanup ensures an allocation is properly closed. It tries to do everything before failing and
 // ensures we don't leave any resources running.
 func (a *Allocation) Cleanup(ctx *actor.Context) {
-	// This message must be sent when the actor is closing since it closes all
-	// websockets listening for these events.
-	exitReason := okExitMessage
-	if a.exitReason != nil {
-		exitReason = a.exitReason.Error()
+	// These messages must be sent when the actor is closing since it
+	// closes all websockets listening for these events.
+	for _, r := range a.exitReasons {
```

Comment:
multiple 'Ended due to ..." logs seems kinda confusing. maybe just one logs and comma separated reasons